src/test/java/com/fasterxml/jackson/databind/introspect/MethodGenericTypeResolverTest.java

```diff
@@ -0,0 +1,196 @@
+package com.fasterxml.jackson.databind.introspect;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.type.TypeBindings;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.junit.Test;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
```

Comment:
Will add a bigger not, but while this could be used in `master` (since 3.0 requires Java 8), it can't be used for 2.12.
Could just use `AtomicReference` as it behaves in many ways the same way, as far as Jackson is concerned.

Reply:
:+1: